spec/validators/status_prerequisite_validator_spec.rb

```diff
@@ -37,6 +37,36 @@
     end
   end
 
+  context "#approved" do
+    let(:error_message) { "approved, requires a draft, a reviewer and for status to be in_review" }
+
+    it "does not allow status to be approved if reviewer_id is missing" do
+      step_by_step_page.reviewer_id = nil
+      step_by_step_page.status = "approved"
+
+      expect(step_by_step_page).not_to be_valid
+      expect(step_by_step_page.errors.messages[:status]).to eq([error_message])
+    end
+
+    it "does not allow status to be approved if there is no draft" do
+      allow(step_by_step_page).to receive(:has_draft?).and_return(false)
+      step_by_step_page.reviewer_id = SecureRandom.uuid
+      step_by_step_page.status = "approved"
+
+      expect(step_by_step_page).not_to be_valid
+      expect(step_by_step_page.errors.messages[:status]).to eq([error_message])
+    end
+
+    it "does not allow status to be approved if the current status is not submitted_for_2i" do
+      allow(step_by_step_page).to receive(:has_draft?).and_return(true)
```

Comment:
I've decided not to change this.  We're using the same step by step for all tests here, so it would result in unnecessary extra code if I were to add an additional create step here.